psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java

```diff
@@ -0,0 +1,513 @@
+            if (generator.getPt() < generator.getPb()) {
+                warnings.add(String.format(Locale.US, "Generator: %d %s Unexpected Pmin: %.2f Pmax: %.2f", generator.getI(), generator.getId(), generator.getPb(), generator.getPt()));
+                validCase = false;
+            }
+            validateGeneratorRegulatingBus(psseBuses, buses, generator);
+
+            addBusesMap(busesGenerators, generator.getI(), generator.getId());
+        }
+
+        Map<String, List<String>> duplicatedBusesGenerators = getDuplicates(busesGenerators);
+        if (!duplicatedBusesGenerators.isEmpty()) {
+            duplicatedBusesGenerators.forEach((key, value) -> warnings
+                .add(String.format("Generator: Multiple generators (%d) at bus %d with the same Id %s", value.size(),
+                    Integer.valueOf(key), value.get(0))));
+            validCase = false;
+        }
+    }
+
+    private void validateGeneratorRegulatingBus(List<PsseBus> psseBuses,  Map<Integer, List<Integer>> buses, PsseGenerator generator) {
+        PsseBus regulatingBus = getRegulatingBus(psseBuses, buses, generator.getIreg(), generator.getI());
+        if (regulatingBus != null
+            && (regulatingBus.getIde() == 2 || regulatingBus.getIde() == 3)
+            && generator.getVs() <= 0.0) {
+            warnings.add(String.format(Locale.US, "Generator: %d %s Unexpected Voltage setpoint: %.2f", generator.getI(), generator.getId(), generator.getVs()));
+            validCase = false;
+        }
+    }
+
+    private void validateNonTransformerBranches(List<PsseNonTransformerBranch> nonTransformerBranches, Map<Integer, List<Integer>> buses) {
+        Map<String, List<String>> busesNonTransformerBranches = new HashMap<>();
+
+        for (int i = 0; i < nonTransformerBranches.size(); i++) {
+            PsseNonTransformerBranch nonTransformerBranch = nonTransformerBranches.get(i);
+            if (!buses.containsKey(nonTransformerBranch.getI())) {
+                warnings.add(String.format("NonTransformerBranch: Unexpected I: %d", nonTransformerBranch.getI()));
+                validCase = false;
+            }
+            if (!buses.containsKey(nonTransformerBranch.getJ())) {
+                warnings.add(String.format("NonTransformerBranch: Unexpected J: %d", nonTransformerBranch.getJ()));
+                validCase = false;
+            }
+            if (nonTransformerBranch.getX() == 0.0) {
+                warnings.add(String.format(Locale.US, "NonTransformerBranch: %d %d %s Unexpected X: %.5f", nonTransformerBranch.getI(), nonTransformerBranch.getJ(), nonTransformerBranch.getCkt(), nonTransformerBranch.getX()));
+                validCase = false;
+            }
+            addBusesMap(busesNonTransformerBranches, nonTransformerBranch.getI(), nonTransformerBranch.getJ(), nonTransformerBranch.getCkt());
+        }
+
+        Map<String, List<String>> duplicatedBusesNonTransformerBranches = getDuplicates(busesNonTransformerBranches);
```

Comment:
can this check for duplicates be made a method that could be used also for two-windings transformers ?

Reply:
Done